### ISSUE TITLE
Replace Lambda env with SSM (or Secrets Manager) for agent token

### ISSUE THREAD

**Issue by pauldraper** [buildkite/elastic-ci-stack-for-aws#597]
Background: While probably not unanimous, there is some consensus that secrets should be kept in SSM parameter store (or Secrets Manager) rather than Lambda env vars.

Lambda functions then use the AWS SDK to request the secrets at runtime.

* https://hackernoon.com/you-should-use-ssm-parameter-store-over-lambda-env-variables-5197fc6ea45b
* https://medium.com/@furqanshaikh/securing-secrets-in-aws-lambda-93c12861371f
* https://dev.to/dvddpl/where-do-you-keep-credentials-for-your-lambda-functions-5dno
* https://www.sethvargo.com/secrets-in-serverless/
* https://www.stackery.io/blog/serverless-secrets/

tl;dr it provides finer-grained access to secrets and lessens likelihood of exposure.

---

Proposal: Have an template paremeter (with default, .e.g. /buildkite/agent-token) for an SSM path. The user is required to place the agent token at this path.

Side benefit: it will fix #422

**Comment by lox** [comment]
This is something we've been discussing a lot recently @pauldraper. I completely agree, although the path to getting there isn't as clear.

The issue is primarily the user experience of creating a new Elastic Stack for the first time. It's incredibly easy to generate a token and create a new stack currently and we're reluctant to impact that. It also makes upgrading between the two approaches hard. 

If the specific issue is that we should be using SSM to pass the secret to lambda, perhaps we could treat that in isolation to entering the token into the template?

**Comment by ianwremmel** [comment]
[`chamber`](https://github.com/segmentio/chamber) (with, in some cases, the assistance of [`aws-vault`](https://github.com/99designs/aws-vault)) makes it really easy to put secrets in a predictable location in SSM.

**Comment by jradtilbrook** [comment]
FWIW I've found that the lambda already supports retrieving a token from SSM by setting the path/key in the environment variable `BUILDKITE_AGENT_TOKEN_SSM_KEY`. So I think the minimal change is to modify this cloudformation template to set that on the lambda.  
I've made a start in #601 and would appreciate review and comments :pray:
